Client/Application/NavigationRouter.swift

```diff
@@ -54,6 +55,8 @@ enum DeepLink {
             self = .homePanel(link)
         } else if component == "default-browser", let link = DefaultBrowserPath(rawValue: String(componentPath)) {
             self = .defaultBrowser(link)
+        } else if component == "trees" {
```

Comment:
Unfortunately when we migrated to our engine we lost our own implementation on deeplinks, but it seems like we could benefit on having an `enum` of possible `hosts` or `components` in a deeplink. At least for having a type safe comparison, if not for having an easier way to switching over all the variants.

Reply:
yes indeed - I was about to wait how the final URL will look like and optimise for type safety afterwards

Reply:
I was giving this a second thought. Having only one custom deeplink, I tend to not refactor the FF deep-linking unless it's actually needed